plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplit.java

```diff
@@ -131,6 +131,33 @@ public HiveSplit(
         this.splitWeight = requireNonNull(splitWeight, "splitWeight is null");
     }
 
+    public HiveSplit withSchema(Properties schema)
```

Comment:
We can't add this without usage.
Since you're looking to reduce the size of `schema` sent in splits, you can take a look at the usage of `schema` in the hive split reading code and limit this to the properties we actually use.
It seems we use only a handful of things like `getDeserializerClassName`, `getInputFormatName`, SKIP_HEADER_COUNT_KEY, SKIP_FOOTER_COUNT_KEY, `name.startsWith("serialization.")` etc.  So we could just cut it down to the actually required things.

Reply:
Will do this on a separate PR